### PR TITLE
Add Feed for Releases and Tags

### LINKED ISSUE #19091: Feed Support for Releases / Tags
as title

basic support for feeds was added at  #569, to get backend information, just look at the 3 related pulls

### PR DESCRIPTION
**Pull request by reopt999**
Fixes #19091

Add Feed for Releases and Tags, can be accessed through `reponame/releases.rss`, `reponame/releases.atom`, `reponame/tags.rss`, and `reponame/tags.atom`

Below is the example of generated feed for rss:
```
<rss version="2.0">
<channel>
<title>Releases for testuser/testrepo</title>
<link>http://localhost:3000/testuser/testrepo/release</link>
<description>This is the test repo</description>
<pubDate>Sun, 06 Nov 2022 14:38:35 +0700</pubDate>
<item>
<title>v1.0</title>
<link>
http://localhost:3000/testuser/testrepo/releases/tag/v1.0
</link>
<description/>
<content:encoded>
&lt;h1 id=&#34;user-content-release-v1-0&#34;&gt;Release v1.0&lt;/h1&gt; &lt;p&gt;&lt;strong&gt;This is the description!&lt;/strong&gt;&lt;br/&gt; Yes&lt;/p&gt;
</content:encoded>
<author>testuser</author>
<guid>
1: http://localhost:3000/testuser/testrepo/releases/tag/v1.0
</guid>
<pubDate>Sun, 06 Nov 2022 14:37:57 +0700</pubDate>
</item>
</channel>
</rss>
```

For atom:
```
<feed>
<title>Releases for testuser/testrepo</title>
<id>http://localhost:3000/testuser/testrepo/release</id>
<updated>2022-11-06T14:43:59+07:00</updated>
<subtitle>This is the test repo</subtitle>
<link href="http://localhost:3000/testuser/testrepo/release"/>
<entry>
<title>v1.0</title>
<updated>2022-11-06T14:37:57+07:00</updated>
<id>
1: http://localhost:3000/testuser/testrepo/releases/tag/v1.0
</id>
<content type="html">
&lt;h1 id=&#34;user-content-release-v1-0&#34;&gt;Release v1.0&lt;/h1&gt; &lt;p&gt;&lt;strong&gt;This is the description!&lt;/strong&gt;&lt;br/&gt; Yes&lt;/p&gt;
</content>
<link href="http://localhost:3000/testuser/testrepo/releases/tag/v1.0" rel="alternate"/>
<summary type="html"/>
<author>
<name>testuser</name>
<email>test@testemail.com</email>
</author>
</entry>
</feed>
```

Below is the specification of the feed header:
- Title: Releases for (name of the repository)
- Link: Link to the list of releases / tags
- Description: Description of the repo
- Created: Time of the feed generated

Below is the specification of the feed items:
- Title: Name of the tag / release
- Link: Link to the release / tag
- Created: Date / time of tag / release creation
- Author: User who created the tag / release
- ID: ID and the link to release / tag
- Content: The description of the release / tag in escaped HTML format

### CODE REVIEW

**Comment by yardenshoham** [comment]
Just a note, we have a PR to add a setting to disable feeds (#21622). When one PR is merged, the other will have to be modified

cc @xin-u

**Comment by silverwind** [comment]
https://github.com/go-gitea/gitea/pull/21622 was merged.

**Comment by reopt999** [comment]
Added support for disable feed with the same configuration as #21622

File: routers/web/feed/convert.go [hunk 1]
```diff
@@ -269,3 +270,47 @@ func GetFeedType(name string, req *http.Request) (bool, string, string) {
...
+			title = rel.Title
+		}
+
+		link := &feeds.Link{Href: rel.HTMLURL()}
+		content, err = markdown.RenderString(&markup.RenderContext{
+			Ctx:       ctx,
+			URLPrefix: rel.Repo.Link(),
+			Metas:     rel.Repo.ComposeMetas(),
+		}, rel.Note)
+		content = html.EscapeString(content)
```
Comment (zeripath) [inline routers/web/feed/convert.go]: This will double-escape the content. The RenderString should safely sanitize.

Comment (zeripath) [reply routers/web/feed/convert.go]:
```suggestion
```

**Review by zeripath — APPROVED** [review]

File: routers/web/feed/release.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
```
Comment (yardenshoham) [inline routers/web/feed/release.go]:
```suggestion
// SPDX-License-Identifier: MIT
```

Different headers were set in #21840

**Comment by silverwind** [comment]
Can we add a RSS icon to releases and tags, similar to how it's done for repos? Might want to wait until https://github.com/go-gitea/gitea/pull/21712 is merged as it does quite a few tweaks around the release HTML.

**Comment by silverwind** [comment]
Thought I guess the feed is for all releases/tags so it's not suitable for display at individual releases/tags.

**Comment by reopt999** [comment]
> Can we add a RSS icon to releases and tags, similar to how it's done for repos?
> 
> image

@silverwind Added RSS icon next to the releases / tags button like below

gscreenshot_2022-12-18-151754

**Review by silverwind — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**